Frameworks/ImageIO/CGImageSource.mm

```diff
@@ -112,6 +115,890 @@ - (CFStringRef)getImageType {
+        // Delay time is represented in 1/100ths of a second, iOS records seconds
+        double delayTime = propertyValue.uiVal/100.0;
+        if (delayTime < 0.1) {
+            [gifProperties setObject:[NSNumber numberWithDouble:0.1] forKey:(id)kCGImagePropertyGIFDelayTime];
+            [gifProperties setObject:[NSNumber numberWithDouble:delayTime] forKey:(id)kCGImagePropertyGIFUnclampedDelayTime];
+        } else {
+            [gifProperties setObject:[NSNumber numberWithDouble:delayTime] forKey:(id)kCGImagePropertyGIFDelayTime];
+            [gifProperties setObject:[NSNumber numberWithDouble:delayTime] forKey:(id)kCGImagePropertyGIFUnclampedDelayTime];
+        }
+    }
+
+    PropVariantClear(&propertyValue);
+    if (SUCCEEDED(imageMetadataReader->GetMetadataByName(L"/logscrdesc/GlobalColorTableFlag", &propertyValue)) && 
+        propertyValue.vt == VT_UI2) {
+        [gifProperties setObject:[NSNumber numberWithInt:propertyValue.uiVal] forKey:(id)kCGImagePropertyGIFHasGlobalColorMap];
+    }
+
+    // Add the GIF dictionary to the properties if there are any GIF properties
+    if ([[gifProperties allKeys] count] != 0) {
+        [properties setObject:gifProperties forKey:(id)kCGImagePropertyGIFDictionary];
+    }
+
+    return (CFDictionaryRef)properties;
+}
+
+CFDictionaryRef readPNGProperties(IWICMetadataQueryReader* imageMetadataReader) {
+    PROPVARIANT propertyValue;
+    PropVariantInit(&propertyValue);
+    NSMutableDictionary* properties = [[NSMutableDictionary alloc] init];
+
+    PropVariantClear(&propertyValue);
+    if (SUCCEEDED(imageMetadataReader->GetMetadataByName(L"/iCCP/ProfileName", &propertyValue)) && propertyValue.vt == VT_LPSTR) {
+        [properties setObject:[NSString stringWithUTF8String:propertyValue.pszVal] forKey:(id)kCGImagePropertyProfileName];
+    }
+
+    // PNG Properties - Format-specific
+    NSMutableDictionary* pngProperties = [[NSMutableDictionary alloc] init];
+
+    PropVariantClear(&propertyValue);
+    if (SUCCEEDED(imageMetadataReader->GetMetadataByName(L"/gAMA/ImageGamma", &propertyValue)) && propertyValue.vt == VT_UI4) {
+        [pngProperties setObject:[NSNumber numberWithInt:propertyValue.ulVal] forKey:(id)kCGImagePropertyPNGGamma];
+    }
+
+    PropVariantClear(&propertyValue);
+    if (SUCCEEDED(imageMetadataReader->GetMetadataByName(L"/sRGB/RenderingIntent", &propertyValue)) && propertyValue.vt == VT_UI1) {
+        [pngProperties setObject:[NSNumber numberWithInt:propertyValue.bVal] forKey:(id)kCGImagePropertyPNGsRGBIntent];
+    }
+
+    PropVariantClear(&propertyValue);
+    if (SUCCEEDED(imageMetadataReader->GetMetadataByName(L"/chrominance/TableEntry", &propertyValue)) && 
+        propertyValue.vt == (VT_VECTOR | VT_UI1)) {
+        // [pngProperties setObject:[NSArray arrayWithObjects:(id)propertyValue.caub.pElems count:propertyValue.caub.cElems] 
+        //                   forKey:(id)kCGImagePropertyPNGChromaticities];
```

Comment:
We weren't handling Chrominance and I'm not sure where the path should actually be. I went ahead and removed Chrominance from source and destination.